crates/sui-types/src/execution.rs

```diff
@@ -169,6 +171,19 @@ impl ExecutionResultsV2 {
                     );
                     *initial_shared_version = lamport_version;
                 }
+
+                // Update initial_shared_version for reshared objects
+                if reshare_at_initial_version {
+                    if let Some(Owner::Shared {
+                        initial_shared_version: previous_initial_shared_version,
+                    }) = input_objects.get(id).map(|obj| &obj.owner)
+                    {
+                        assert!(!self.created_object_ids.contains(id));
```

Comment:
is it worth removing the assert and making this an if/else?

Reply:
true, but if we think that's worth an assertion shouldn't it apply to all owner types? that said, i'm not very worried about this either way